sdks/java/extensions/sql/perf-tests/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/bigquery/BigQueryIOPushDownIT.java

```diff
@@ -0,0 +1,198 @@
+import org.apache.beam.sdk.io.common.IOITHelper;
+import org.apache.beam.sdk.io.common.IOTestPipelineOptions;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.testutils.NamedTestResult;
+import org.apache.beam.sdk.testutils.metrics.IOITMetrics;
+import org.apache.beam.sdk.testutils.metrics.MetricsReader;
+import org.apache.beam.sdk.testutils.metrics.TimeMonitor;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.Row;
+import org.apache.beam.vendor.calcite.v1_20_0.org.apache.calcite.plan.RelOptRule;
+import org.apache.beam.vendor.calcite.v1_20_0.org.apache.calcite.tools.RuleSet;
+import org.apache.beam.vendor.calcite.v1_20_0.org.apache.calcite.tools.RuleSets;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class BigQueryIOPushDownIT {
+  private static final String READ_FROM_TABLE = "bigquery-public-data:hacker_news.full";
+  private static final String NAMESPACE = BigQueryIOPushDownIT.class.getName();
+  private static final String CREATE_TABLE_STATEMENT =
+      "CREATE EXTERNAL TABLE HACKER_NEWS( \n"
+          + "   title VARCHAR, \n"
+          + "   url VARCHAR, \n"
+          + "   text VARCHAR, \n"
+          + "   dead BOOLEAN, \n"
+          + "   `by` VARCHAR, \n"
+          + "   score INTEGER, \n"
+          + "   `time` INTEGER, \n"
+          + "   `timestamp` TIMESTAMP, \n"
+          + "   type VARCHAR, \n"
+          + "   id INTEGER, \n"
+          + "   parent INTEGER, \n"
+          + "   descendants INTEGER, \n"
+          + "   ranking INTEGER, \n"
+          + "   deleted BOOLEAN \n"
+          + ") \n"
+          + "TYPE 'bigquery' \n"
+          + "LOCATION '"
+          + READ_FROM_TABLE
+          + "' \n"
+          + "TBLPROPERTIES '{ method: \"%s\" }'";
+  private static final String SELECT_STATEMENT =
+      "SELECT `by` as author, type, title, score from HACKER_NEWS where (type='story' or type='job') and score>2";
+
+  private static SQLBigQueryPerfTestOptions options;
+  private static String metricsBigQueryDataset;
+  private static String metricsBigQueryTable;
+  private Pipeline pipeline = Pipeline.create(options);
+  private BeamSqlEnv sqlEnv;
+
+  @BeforeClass
+  public static void setUp() {
+    options = IOITHelper.readIOTestPipelineOptions(SQLBigQueryPerfTestOptions.class);
+    metricsBigQueryDataset = options.getMetricsBigQueryDataset();
+    metricsBigQueryTable = options.getMetricsBigQueryTable();
+  }
+
+  @Before
+  public void before() {
+    sqlEnv = BeamSqlEnv.inMemory(new BigQueryPerfTableProvider(NAMESPACE, "fields_read"));
+  }
+
+  @Test
+  public void readUsingDirectReadMethodPushDown() {
+    sqlEnv.executeDdl(String.format(CREATE_TABLE_STATEMENT, "DIRECT_READ"));
```

Comment:
You should make `"DIRECT_READ"` an constant.

Reply:
Ended up pooling `String` value for `DIRECT_READ` and `DEFAULT` from `enum Method` in `BigQueryIO`.